adobe_vipm/flows/fulfillment/shared.py

```diff
@@ -1002,95 +1004,149 @@ class CreateOrUpdateSubscriptions(Step):
-                else:
-                    adobe_sku = line["offerId"]
-                    set_subscription_actual_sku(
-                        client,
-                        context.order,
-                        order_subscription,
-                        adobe_sku,
-                    )
-                    logger.info(
-                        "%s: subscription %s (%s) updated",
-                        context,
-                        line["subscriptionId"],
-                        order_subscription["id"],
-                    )
         next_step(client, context)
 
+    def _update_existing_subscription(self, client, context, order_subscription, line):
+        """Update an existing subscription with new Adobe SKU."""
+        adobe_sku = line["offerId"]
+        set_subscription_actual_sku(
+            client,
+            context.order,
+            order_subscription,
+            adobe_sku,
+        )
+        logger.info(
+            "%s: subscription %s (%s) updated",
+            context,
+            line["subscriptionId"],
+            order_subscription["id"],
+        )
+
+    def _create_new_subscription(self, client, context, adobe_client, line, order_line):
+        """Create a new subscription."""
+        adobe_subscription = self._get_adobe_subscription(adobe_client, context, line)
+        if not adobe_subscription:
+            return
+
+        template = self._get_subscription_template(client, context)
+        subscription_data = self._build_subscription_data(
+            line, order_line, adobe_subscription, template
+        )
+
+        subscription = create_subscription(client, context.order_id, subscription_data)
+        logger.info(
+            "%s: subscription %s (%s) created",
+            context,
+            line["subscriptionId"],
+            subscription["id"],
+        )
+
+    def _get_adobe_subscription(self, adobe_client, context, line):
+        """Get Adobe subscription and validate its status."""
+        adobe_subscription = adobe_client.get_subscription(
+            context.authorization_id,
+            context.adobe_customer_id,
+            line["subscriptionId"],
+        )
+
+        if adobe_subscription["status"] != AdobeStatus.PROCESSED:
+            logger.warning(
+                "%s: subscription %s for customer %s is in status %s, skip it",
+                context,
+                adobe_subscription["subscriptionId"],
+                context.adobe_customer_id,
+                adobe_subscription["status"],
+            )
+            return None
+
+        return adobe_subscription
+
+    def _get_subscription_template(self, client, context):
+        """Get subscription template with error handling."""
+        try:
+            return get_template_by_name(
+                client,
+                context.order["agreement"]["product"]["id"],
+                TEMPLATE_SUBSCRIPTION_AUTORENEWAL_ENABLE,
+            )
+        except Exception:
```

Comment:
I'm not really sure, that we need to have here exception handler... We assume that template exists or at least default template exists